tests/blackbox_test.py

```diff
@@ -0,0 +1,629 @@
+import os
+from pathlib import Path
+from itertools import product
+from typing import Dict, Tuple
+
+import pytest
+
+from .compile_asserts import assert_teal_as_expected
+from .blackbox_asserts import (
+    algod_with_assertion,
+    blackbox_pyteal,
+    mode_to_execution_mode,
+)
+
+from algosdk.testing.dryrun import Helper as DryRunHelper
+from algosdk.testing.teal_blackbox import (
+    DryRunEncoder as Encoder,
+    DryRunExecutor,
+    DryRunProperty as DRProp,
+    DryRunTransactionResult,
+    SequenceAssertion,
+)
+
+from pyteal import *
+
+# TODO: get tests working on github and set this to True
```

Comment:
@tzaffi FYI - I'm _trying_ to work on the TODO for integration tests on CI.

I'm working out of https://github.com/algorand/pyteal/pull/260.  If it proves too challenging, I may suggest opening a separate issue.

Reply:
Sync'ed up offline but let's summarize my wish list regarding testing (with the caveat that clearly "testing" is better than "no testing").

From most desired to least desired (bug again, even least desired is really great):
1. have a dry run test that doesn't require running a node (a la `goal clerk dryrun` without the `remote` option)
2. have the dry run test that uses a quickly downloadable pre-built docker image (a la `docker pull algorand/stable`)
3. have a dry run that stands up the minimum necessary sandbox-like environment
4. have a dry run that stands up a "heavier" sandbox environment similar to the one that the SDK's do

Reply:
resolving as this is a separate issue discussed on #260